File: TiemposOrdenamiento/C++/seleccion.cpp
#include<iostream>
#include <time.h>
#include <stdio.h>
#include <dos.h> 
#include<chrono>
using namespace std;

int main() {
	int i, n, *a;
	
	for (n=10; n<=500; n+=10){
	
		a = new int[n];
		
		//relleno
		for(i=0; i<n; i++){
			a[i] = i+1;
		}
		
		//ordenamiento
		int j, min;

		auto start = std::chrono::high_resolution_clock::now();
		for (int i = 0; i < n - 1; i++) {
			min = i;
			for (int j = i + 1; j < n; j++) {
				if (a[j] < a[min]) { // comparacion
					min = j;
				}
			}
			// intercambio
			int aux = a[min];
			a[min] = a[i];
			a[i] = aux;
		}
		auto finish = std::chrono::high_resolution_clock::now();
		cout << "N=" <<n<<", tiempo: "; 
	    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(finish-start).count() << "ns\n";
	    cout << endl;
	    
	}
	
	    
	//impresion
	for(i=0; i<n; i++){
		//cout<<a[i]<<",";
	}
	
	delete[] a;
}

File: TiemposOrdenamiento/Java/Ordenamiento/src/ordenamiento/Seleccion.java
package ordenamiento;

public class Seleccion {

	public static void main(String[] args) {

		int[] a;

		for (int n = 10; n <= 500; n += 10) {

			a = new int[n];

			// relleno
			for (int i = 0; i < n; i++) {
				a[i] = i + 1;
			}

			int min;

			// ordenamiento
			long t0 = System.nanoTime();
			for (int i = 0; i < n - 1; i++) {
				min = i;
				for (int j = i + 1; j < n; j++) {
					if (a[j] < a[min]) { // comparacion
						min = j;
					}
				}
				// intercambio
				int aux = a[min];
				a[min] = a[i];
				a[i] = aux;
			}
			long t1 = System.nanoTime();

			// impresion
			System.out.println("N=" + n + ", tiempo: " + (t1 - t0) + "ns");
			for (int i : a) {
				// System.out.print(i + ",");
			}
		}
	}

}
